lib/psci/psci_common.c

```diff
@@ -395,6 +395,8 @@ void psci_do_state_coordination(unsigned int end_pwrlvl,
 	plat_local_state_t target_state, *req_states;
 
 	assert(end_pwrlvl <= PLAT_MAX_PWR_LVL);
+	if (end_pwrlvl > PLAT_MAX_PWR_LVL)
+		return;
```

Comment:
As stated [here](https://github.com/ARM-software/arm-trusted-firmware/pull/1058#issuecomment-323919722), the assert can be removed if a runtime check is added. Also, since this case should never happen, it would be better to panic() instead of returning. A comment to explain this would also help.

Actually, given that this is a false positive, I would prefer a solution that didn't result in additional code in the final binary. For example, a compiler hint to suppress the warning. But I don't know if that is possible.